src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -26,6 +26,7 @@ public unsafe static uint[] Square(uint[] value)
 
         // Mutable for unit testing...
         private static int SquareThreshold = 32;
+        private static int AllocationThreshold = 4096;
```

Comment:
Nope, it's array length. It's just a wild guess and I hoped someone has a more profound suggestion. The idea is that numbers around 8 KB can live on the stack, 4096 means 16 KB values and bigger use array allocations instead.